state/multiwatcher/multiwatcher.go

```diff
@@ -272,6 +292,7 @@ type UnitInfo struct {
 	MachineId      string      `json:"machine-id"`
 	Ports          []Port      `json:"ports"`
 	PortRanges     []PortRange `json:"port-ranges"`
+	Principal      string      `json:"principal"`
```

Comment:
Is it worth adding a test to ensure that when non-empty, this materialises in watchers?

Reply:
I'd say no, it doesn't have an omit empty on it and an empty string is okay.